### PR TITLE
Start adding CSV tests

### PR DESCRIPTION
**Pull request by thbar**
This is just a draft to open discussion to implement CSV tests (#284). As I worked on it I had a few a-ha moments (partly due to my discovering of the `from_csv` API).

I used https://github.com/jorgecarleitao/arrow2/tree/main/src/io/csv/read as an initial basis.

I used a data-based approach to iterate and generate tests using `quote`/`unquote`, something I found useful to discover the API, but maybe we'll want something more explicit & more readable (e.g. one describe block per "dtype", manually written).

I have not attempted to cover in full what can be done, partly due to personal bandwidth, and also to get feedback.

Things I have noticed/questions I have that could be of interest to continue this:
- the `quote/unquote` implementation is not very readable, but makes it easy to add an extra sample
- My understanding is that Polars is the default backend, and `from_csv` goes via the backend, so presumably we'll want something that can be made backend-agnostic but still run on all the backends? (I haven't looked much into the backends code at this point)
- There would be a lot of interest to add tests around many more things: a misparsed datetime will result in nil (I believe), is the library failing fast on errors such as a random string in the middle of an int column with an explicitly passed dtype, how does float truncation occurs (I believe extra digits in the CSV column will be just lost at this point), etc etc
- `frame["city"] == frame[0]` returns false (a different reference) and I presume this could be linked to the Rust interop, but haven't looked into it yet

### CODE REVIEW

**Comment by josevalim** [comment]
> The quote/unquote implementation is not very readable, but makes it easy to add an extra sample

I think you can use functions without loss of behaviour:

    assert_csv :datetime, "\"string\"", "string"

> My understanding is that Polars is the default backend, 

So far we only test the default backend, so I would say this approach is fine!

> frame["city"] == frame[0] returns false

Correct, you want to use Series.equal or similar. Another test you want to consider adding is that we can decode and then encode the CSV and it goes back to the same value. :)

**Review by josevalim — APPROVED** [review]

**Comment by thbar** [comment]
Good enough for a first round, thanks @josevalim for the suggestions & review!

I will likely come back with edge-cases (round-trip giving different outcome, what happens with digits/precision, error handling etc).

File: test/explorer/data_frame/csv_test.exs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      assert_csv(:date, "2022-12-01", ~D[2022-12-01])
+      assert_csv(:date, "1960-01-31", ~D[1960-01-31])
+    end
+
+    test "datetime" do
+      assert_csv(:datetime, "2022-10-01T11:34:10.123456", ~N[2022-10-01 11:34:10.123456])
+    end
+
+    @tag :skip
+    test "list"
```
Comment (josevalim) [inline test/explorer/data_frame/csv_test.exs:105]:
```suggestion
```

**Review by philss — APPROVED** [review]

**Comment by philss** [comment]
@thbar thank you! :purple_heart:
